### DISCUSSION TITLE
Any plans to apply retention policy to VolumeSnapshot backups?

### DISCUSSION THREAD

**Discussion by pkutakov** [cloudnative-pg/cloudnative-pg#6030] [General]
Is there any plans like this? Do you see some blocking issues from snapshotting mechanism that prevents it to be happen?

**Comment by Infinoid** [comment]
Now that [backup retention policies are deprecated](https://cloudnative-pg.io/documentation/1.27/backup/#retention-policies) in favor of plugins (and csi snapshots are not configured via plugin), I assume not.

I had some related questions: 

* If I delete old `Backup` objects by hand, or implement some other mechanism to clean them up, should `cnpg status` recognize that and adjust the "First Point Of Recoverability" to match?  (In v1.27, looks like it doesn't.)
* When I delete a `Backup` object, shouldn't it delete the corresponding `VolumeSnapshot` object automatically?  (In v1.27, looks like it doesn't.)
* If I have snapshot backups and (barman-plugin-based) WAL archiving configured, how closely do the two retention policies need to be kept in sync?  Is it just wasting space to store WAL archives older than the oldest snapshot, or can larger problems occur?
* Am I better off using the barman plugin for backups, too?  Snapshots don't seem very well supported at the moment.

**Comment by vadius45** [comment]
volumesnapshots and the barman plugin for WALs are complementary, and I also have the same question about the retention policy for snapshots. Seems handy and consistent that it's enforced by CNPG.

**Comment by Infinoid** [comment]
For now, I run https://gist.github.com/Infinoid/683e148cd5808c1c1c2563eb99ca8b94 as a CronJob.  It's not very pretty, or smart (it doesn't even check whether your backups are snapshot-based), but it seems to delete some things.

Disclaimers: this code is not official or supported by the cloudnative-pg project, it's just something I bashed together, do not run code you find on the ground without reviewing it first, don't blame me if it does something wrong, including (but not limited to) stabbing your kittens.
